drone/src/agent/docker.rs

```diff
@@ -193,19 +193,31 @@ impl DockerInterface {
         )
     }
 
-    pub fn get_stats(
-        &self,
-        container_name: &str,
-    ) -> impl Stream<Item = Result<Stats, bollard::errors::Error>> {
+    /// The docker api (as of docker version 20.10.18) blocks for ~1s before returning
+    /// from self.docker.stats, hence the effective minimal interval is a second
+    pub fn get_stats<'a>(
+        &'a self,
+        container_name: &'a str,
+    ) -> impl Stream<Item = Result<Stats, bollard::errors::Error>> + 'a {
         let options = StatsOptions {
-            stream: true,
-            one_shot: false,
+            stream: false,
+            one_shot: true,
         };
-        self.docker
-            .stats(container_name, Some(options))
-            .throttle(std::time::Duration::from_secs(
-                DEFAULT_DOCKER_THROTTLED_STATS_INTERVAL_SECS,
-            ))
+
+        IntervalStream::new({
```

Comment:
also, I was wondering, how does rust know to drop this stream? does it do it when  self gets dropped?

Reply:
Interesting, is there any way to define a 'container lifetime that'll warn about stuff like this?